presto-hive/src/main/java/com/facebook/presto/hive/HiveTableHandle.java

```diff
@@ -49,34 +63,41 @@ public String getTableName()
         return tableName;
     }
 
-    public SchemaTableName getSchemaTableName()
+    @JsonProperty
+    public Optional<HiveAnalyzePropertiesHandle> getAnalyzeHandle()
     {
-        return new SchemaTableName(schemaName, tableName);
+        return analyzeHandle;
     }
 
-    @Override
-    public int hashCode()
+    public SchemaTableName getSchemaTableName()
     {
-        return Objects.hash(schemaName, tableName);
+        return new SchemaTableName(schemaName, tableName);
     }
 
     @Override
-    public boolean equals(Object obj)
+    public boolean equals(Object o)
     {
-        if (this == obj) {
+        if (this == o) {
             return true;
         }
-        if (obj == null || getClass() != obj.getClass()) {
+        if (o == null || getClass() != o.getClass()) {
             return false;
         }
-        HiveTableHandle other = (HiveTableHandle) obj;
-        return Objects.equals(this.schemaName, other.schemaName) &&
-                Objects.equals(this.tableName, other.tableName);
+        HiveTableHandle that = (HiveTableHandle) o;
+        return Objects.equals(schemaName, that.schemaName) &&
+                Objects.equals(tableName, that.tableName) &&
+                Objects.equals(analyzeHandle, that.analyzeHandle);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(schemaName, tableName, analyzeHandle);
     }
 
     @Override
     public String toString()
     {
-        return schemaName + ":" + tableName;
+        return schemaName + ":" + tableName + ":" + analyzeHandle;
```

Comment:
This `toString()` is used when printing the query plan. We probably don't want to include the partition values.

Reply:
I think we may want to print it. 

If the user is analyzing the whole table, the partition values filed here will be empty so nothing would be printed. 

If the user specify a list of partitions to analyze, having them in the plan can help user verify if we actually analyzed those partitions. The plan will be as long as the analyze query itself. 

An alternative way is to print up to N partitions here.